app/controllers/publish/providers/accredited_partnerships/checks_controller.rb

```diff
@@ -5,16 +5,21 @@ module Providers
     module AccreditedPartnerships
       class ChecksController < ApplicationController
         def show
-          provider_partnership_form
+          @partnership = provider.accredited_partnerships.build(accredited_provider:)
+
+          if @partnership.invalid?
+            redirect_to search_publish_provider_recruitment_cycle_accredited_providers_path(provider.provider_code, recruitment_cycle.year), flash: { error: { message: "#{accredited_provider.name_and_code} partnership already exists" } }
+          end
         end
 
         def update
-          @partnership = provider.accredited_partnerships.build(accredited_provider: provider_partnership_form.accredited_provider, description: provider_partnership_form.description)
+          @partnership = provider.accredited_partnerships.build(accredited_provider:)
 
           if @partnership.save
             notify_accredited_provider_users
 
-            redirect_to publish_provider_recruitment_cycle_accredited_partnerships_path(@provider.provider_code, @provider.recruitment_cycle_year), flash: { success: t(".added") }
+            flash[:success_with_body] = { "title" => t(".added"), "body" => accredited_provider.provider_name }
+            redirect_to publish_provider_recruitment_cycle_accredited_partnerships_path(@provider.provider_code, @provider.recruitment_cycle_year)
```

Comment:
Personally, I prefer to tie these together.

```suggestion
            redirect_to publish_provider_recruitment_cycle_accredited_partnerships_path(@provider.provider_code, @provider.recruitment_cycle_year), flash: { success_with_body: { "title" => t(".added"), "body" => accredited_provider.provider_name } }
```

Reply:
I did until Lori suggested otherwise and I tend to agree with her. Putting it on another line makes it more of a first class operation. Tacking it onto the end of the redirect makes it feel less important.

Also, the `success_with_body` is long